anxcloud/resource_virtual_server.go

```diff
@@ -290,33 +289,17 @@ func resourceVirtualServerRead(ctx context.Context, d *schema.ResourceData, m in
 		return nil
 	}
 
-	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```

Comment:
Should we keep this code active for VMs that are not `from_scratch`?

Reply:
This retry also breaks `templates` typed VMs when the VLAN is empty (no IP from VLAN assigned to VM).

Reply:
I think the user could remove the IP address from the NIC and affect the info endpoint. Then the provider would have no way to check if it's really an empty VLAN.

Reply:
Mhmm I see ... I'm worried we might run into a problem like freshly creating a VM and using its IPs in other resources, but since we aren't waiting for them to be reported, we don't have them

Something like this:
```hcl
resource "anxcloud_virtual_server" "servicevm" {
    // ...
}

resource "anxcloud_dns_record" "servicevm_record" {
  name      = anxcloud_virtual_server.servicevm.hostname
  zone_name = "example.com"
  type      = "A"
  rdata     = anxcloud_virtual_server.servicevm.info.network[0].ip_v4
}
```

But after writing this, I have a feeling like we cannot use the IP from the info attribute for something like this anyway? Is there a scenario where we don't allocate an IP via terraform but want to use an IP the VM has for other terraform things?

Reply:
This wasn't reliably possible before, because of a bug where the data from the info endpoint isn't updated correctly
https://github.com/anexia-it/terraform-provider-anxcloud/pull/51/files#diff-6bc440c6ee6b20a283242a9b6293c74f9389d45b4b06c1817467cc33a6659b30R293

I'll create a ticket regarding this issue.